src/frontend-samples/issues-sample/IssuesWidget.tsx

```diff
@@ -189,7 +193,12 @@ const IssuesWidget: React.FunctionComponent = () => {
         setCurrentIssue(iss);
       });
     }
-  }, [applyView, currentIssues]);
+    /** This timer is being used to resolve an async issue that prevents marker pin decorations from being drawn immeditately after changing the current issues */
+    const timer = new Timer(250);
+    timer.setOnExecute(() => { if (IModelApp.viewManager) IModelApp.viewManager.invalidateDecorationsAllViews(); });
+    timer.start();
```

Comment:
Really not a fan of this and there has to be a better way. We just aren't doing something right. I've played around with it for a few minutes but i couldn't solve it trivially. It's like its just not rendering the next frame after changing the decoration. Maybe we should reach out to someone for help.